stopWatch.html

```diff
@@ -1,326 +1,352 @@
+        "<div class='row'>" +
+        "<div class='col-md-8'>" +
+        "<h3 class='panel-title'>" +
+        Watch.title +
+        "</h3>" +
+        "</div>" +
+        "<div class='col-md-4 text-center'>" +
+        "<button type='button' class='btn btn-default glyphicon glyphicon-pencil' data-toggle='modal' data-target='#notesModal' onclick='fillModal(" +
+        Watch.id + ")'>" +
+        " Notes" +
+        "</button>" +
+        "</div>" +
+        "</div>" +
+        "</div>" +
+        "<div class='panel-body'>" +
+        "<h2 class='text-center' + id='watch" + Watch.id + "'>" +
+        GiveTimeString(Watch.curTime - Watch.startTime) +
+        "</h2>" +
+        "</div>" +
+        "<div class='panel-footer'>" +
+        "<div class='btn-group btn-group-justified'>" +
+        pausePlayButtonStr(Watch.status) +
+        "<span class='btn btn-info glyphicon glyphicon-refresh' onclick='RestartClock(" + Watch.id +
+        ")'><span class='btn-text'>Restart</span></span>" +
+        "<span class='btn btn-danger glyphicon glyphicon-remove' onclick='RemoveOne(" + Watch.id +
+        ")'><span class='btn-text'>Remove</span></span>" +
+        "<!/div>" +
+        "</div>" +
+        "</div>" +
+        "</div>";
+
+      return retStr;
+    }
+
+    //creates a watch object
+    function Watch(status = 0, title, startTime = Date.now(), curTime = Date.now(), notes = []) {
+      if (title == null)
+        title = document.getElementById('title').value;
+      this.id = idCounter;
+      idCounter = idCounter + 1;
+      this.status = status; // 0 -> pause state 1 -> play state
+      this.title = title;
+      this.startTime = startTime;
+      this.curTime = curTime;
+      this.notes = notes;
+    }
+
+    //adds watch to DOM
+    function AddWatch() {
+      listStopWatch[listStopWatch.length] = new Watch();
+      $("#stopwatches").append(StopWatchBody(listStopWatch[listStopWatch.length - 1]));
+      document.getElementById('title').value = '';
+    }
+
+    function ShowList() {
+      console.log(listStopWatch);
+      // for(var i = 0; i < listStopWatch.length; i++) {
+      // 	console.log(listStopWatch[i].title);
+      // }
+    }
+
+    //removes a single clock when called
+    function RemoveOne(id) {
+      document.getElementById(id).outerHTML = '';
+      for (var i = 0; i < listStopWatch.length; i++) {
+        if (listStopWatch[i].id == id) {
+          listStopWatch.splice(i, 1);
+          break;
+        }
+      }
+    }
+
+    //restarts the time for one clock
+    function RestartClock(id) {
+      for (var i = 0; i < listStopWatch.length; i++) {
+        if (listStopWatch[i].id == id) {
+          listStopWatch[i].startTime = Date.now();
+          listStopWatch[i].curTime = Date.now();
+          document.getElementById('watch' + id).innerHTML = GiveTimeString(0);
+        }
+      }
+    }
+
+    //for pause play
+    function PausePlayToggle(elem, id) {
+      if (elem.textContent == 'Pause') {
```

Comment:
what's the benefit of using '.textContent' instead of '.innerHTML' here.

Reply:
The buttons (Play/Pause, Restart, Remove) have no padding between the icon images and the text. In order to add padding between the icon and the text on those buttons, I had to place spans around the text. 

When placing spans around the Play/Pause text, it breaks the PausePlayToggle function because that function updates the functionality of the button based on the text content of the button. Originally that function matched based on the innerHTML. However, the innerHTML of that button then becomes "<span...>Play</span>", which doesn't match 'Play'. 

Matching on innerHTML means that adding the inner spans for padding causes the Play/Pause button to stop working since the function can no longer find it in the DOM. By matching based on text content instead, the function does not break and is also more scalable. You can update the HTML of the button and it will still work providing the text content remains 'Play' or 'Pause'.

Reply:
Please let me know if I'm not making sense in my explanation. It's kind of a weird thing to describe. Also, in this PR, the padding I added between the text and icons in the buttons is really small (2px), which is maybe why it wasn't so noticeable.